tests/test_integrated_channels/test_cornerstone/test_utils.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the utilities used by Cornerstone integration channel.
+"""
+
+from __future__ import absolute_import, unicode_literals, with_statement
+
+import unittest
+
+import ddt
+from pytest import mark
+
+from django.test import RequestFactory
+
+from integrated_channels.cornerstone.models import CornerstoneLearnerDataTransmissionAudit
+from integrated_channels.cornerstone.utils import create_cornerstone_learner_data_transmission_audit
+from test_utils.factories import UserFactory
+
+
+@ddt.ddt
+class TestCornerstoneUtils(unittest.TestCase):
+    """
+    Test utility functions used by Cornerstone integration channel.
+    """
+    @staticmethod
+    def _assert_learner_data_transmission_audit(transmission_audit, querystring):
+        """ Asserts querystring params with CornerstoneLearnerDataTransmissionAudit """
+        assert transmission_audit.user_guid == querystring['userGuid']
+        assert transmission_audit.course_id == querystring['courseId']
+        assert transmission_audit.session_token == querystring['sessionToken']
+        assert transmission_audit.callback_url == querystring['callbackUrl']
+        assert transmission_audit.subdomain == querystring['subdomain']
+
+    @staticmethod
+    def _get_request(querystring, user=None):
+        """ returns mocked request """
+        request = RequestFactory().get(path='/', data=querystring)
+        request.user = user if user else UserFactory()
+        return request
+
+    @ddt.data(
+        (
+            {
+                'userGuid': 'dummy_id',
+                'courseId': 'dummy_courseId',
+                'sessionToken': 'dummySessionToken',
+                'callbackUrl': 'dummy_callbackUrl',
+                'subdomain': 'dummy_subdomain',
+            },
+            True,
+        ),
+        (
+            {
+                'userGuid': 'dummy_id',
+                'callbackUrl': 'dummy_callbackUrl',
+                'subdomain': 'dummy_subdomain',
+            },
+            False,
+        ),
+        (
+            {},
+            False,
+        ),
+    )
+    @ddt.unpack
+    @mark.django_db
+    def test_update_cornerstone_learner_data_transmission_audit(self, querystring, expected_result):
+        """ test creating records """
+        request = self._get_request(querystring)
+        create_cornerstone_learner_data_transmission_audit(request)
+        assert CornerstoneLearnerDataTransmissionAudit.objects.exists() == expected_result
```

Comment:
I think to make this assertion safe you should pull `cornerstone_transmission_audit` records of requesting user instead of pulling all entries in `CornerstoneLearnerDataTransmissionAudit`. e.g if any other operation in the test case created an entry for  `CornerstoneLearnerDataTransmissionAudit` that can pass this assertion.